src/models/src/entity/clients.rs

```diff
@@ -1110,13 +1110,13 @@ pub fn is_origin_external<'a>(
     let scheme_ok = if *PROXY_MODE && scheme == "https" {
         true
     } else {
-        match listen_scheme {
+        (match listen_scheme {
             ListenScheme::Http => scheme == "http",
             ListenScheme::Https => scheme == "https",
             ListenScheme::HttpHttps => scheme == "http" || scheme == "https",
             ListenScheme::UnixHttp => scheme == "http",
             ListenScheme::UnixHttps => scheme == "https",
-        }
+        } || ADDITIONAL_ALLOWED_ORIGIN_SCHEMES.contains(&scheme.to_string()))
```

Comment:
This would be cleaner if integrated into the `match` directly as a catch-all like `x => ...` and no conversion with `to_string()` to avoid allocation, but use a reference.

Reply:
Sadly the contains function is not happy receiving a &str, but the allocation can be resolved by using 
ADDITIONAL_ALLOWED_ORIGIN_SCHEMES.iter().any(|s| s.as_str() == scheme)

I don't think it can exist within the match statement, as the listenscheme is from the server itself and would make it unreachable.

Reply:
You're right of course, yes.